### ISSUE TITLE
Attaching empty files to Test with DefectDojo persistence provider

### ISSUE THREAD

**Issue by EndPositive** [secureCodeBox/secureCodeBox#877]
### Describe the bug
In some cases, scanners may output empty raw result files (e.g. Nuclei jsonl). When importing these into DefectDojo with the DefectDojo persistence provider, the hook fails with: `Failed to attach findings to engagement.`. DefectDojo reports a Bad Request. Uploading the same raw file in DefectDojo UI reports that the findings file may not be empty.

### Steps To Reproduce
With DD, DD persistence, and Nuclei installed, run a Nuclei scan in SCB without any endpoints specified. This results in an empty raw result file for Nuclei.

### Expected behavior
DefectDojo Test is made without any imported results.

### System (please complete the following information):
 - secureCodeBox Version/Release: 3.5.0
 - DefectDojo Version: 2.4.1

### Screenshots / Logs
```
[13/Dec/2021 09:18:08] WARNING [django.request:224] Bad Request: /api/v2/reimport-scan/
WARNING:django.request:Bad Request: /api/v2/reimport-scan/
[pid: 27|app: 0|req: 51/78] 172.17.0.1 () {40 vars in 727 bytes} [Mon Dec 13 09:18:08 2021] POST /api/v2/reimport-scan/ => generated 130 bytes in 12 msecs (HTTP/1.1 400) 7 headers in 203 bytes (1 switches on core 1)
```
```
2021-12-13 09:17:51 INFO  DefectDojoPersistenceProvider:24 - Starting DefectDojo persistence provider
2021-12-13 09:17:53 INFO  DefectDojoPersistenceProvider:35 - Downloading Scan Result
2021-12-13 09:17:55 INFO  DefectDojoPersistenceProvider:39 - Uploading Findings to DefectDojo at: http://defectdojo-django.default.svc.cluster.local
2021-12-13 09:17:57 INFO  VersionedEngagementsStrategy:87 - Running with DefectDojo User Id: 1
2021-12-13 09:17:57 INFO  VersionedEngagementsStrategy:200 - Looking for ID of ProductType 'Research and Development'
2021-12-13 09:17:57 INFO  VersionedEngagementsStrategy:207 - Using ProductType Id: 1
Exception in thread "main" io.securecodebox.persistence.defectdojo.exceptions.DefectDojoPersistenceException: Failed to attach findings to engagement.
	at io.securecodebox.persistence.defectdojo.service.ImportScanService.createFindings(ImportScanService.java:136)
	at io.securecodebox.persistence.defectdojo.service.ImportScanService.reimportScan(ImportScanService.java:151)
	at io.securecodebox.persistence.strategies.VersionedEngagementsStrategy.run(VersionedEngagementsStrategy.java:103)
	at io.securecodebox.persistence.DefectDojoPersistenceProvider.main(DefectDojoPersistenceProvider.java:42)
```

**Comment by J12934** [comment]
Does DefectDojo report anything back in the response body of the request that we could catch?
Or just empty body with response code 400?

**Comment by EndPositive** [comment]
This is the response body:

```
{'file': ['The submitted file is empty.'], 'message': "{'file': [ErrorDetail(string='The submitted file is empty.', code='empty')]}"}
```

**Comment by EndPositive** [comment]
Maybe you could also log the response body for failed requests in the DD hook.

E.g. here: https://github.com/secureCodeBox/defectdojo-client-java/blob/1e4f0338bb78332ee488710ec5c6d5f93bb2325a/src/main/java/io/securecodebox/persistence/defectdojo/service/ImportScanService.java#L136

**Comment by J12934** [comment]
Nice the error is pretty explicit, should be possible to catch this to exit gracefully as this really isn't a error case for us.

Logging the response out when the requests fails would probably be good in general for the PersistenceProvider, not just for the finding upload. 👍

**Comment by damiencarol** [comment]
Hello guys, just saw this one. If the tool can produce empty report files, DefectDojo must support it.
We try to support the report as-is.
Do you have an example of empty report? Which tools generate empty files?

**Comment by EndPositive** [comment]
Hi @damiencarol, nuclei can generate empty reports. It uses the jsonl format, so if nuclei reports no findings, the report is empty.

**Comment by damiencarol** [comment]
I will add a unit test for this use case. Thanks for your answer. I really think it's a bug in DD.
Will fix it for `2.7.0`

**Comment by damiencarol** [comment]
I'm able to reproduce the error.
```
(.venv2) [damien@damien dd2]$ python3 test_empty_report.py 
b'{"file":["The submitted file is empty."],"message":"{\'file\': [ErrorDetail(string=\'The submitted file is empty.\', code=\'empty\')]}"}'
<Response [400]>
400 Client Error: Bad Request for url: https://demo.defectdojo.org/api/v2/import-scan/
```
working on a fix.

**Comment by damiencarol** [comment]
Fixed in `2.7.0`

**Comment by Ilyesbdlala** [comment]
This issue is no longer encountered as of at least SCB `v3.10.0`.
